Software/lib/GNSE_HAL/GNSE_hal.h

```diff
@@ -42,10 +42,21 @@ typedef struct GNSE_HAL_Ctx_s
     bool buzzer_init;
 } GNSE_HAL_Ctx_t;
 
+/**
+ * Clock return types
+ */
+typedef enum
+{
+    CLK_OP_SUCCESS = 0,
+    CLK_OP_FAIL = 1,
+} CLK_op_result_t;
```

Comment:
We need to have `GNSE_HAL_op_result_t;` instead of per peripheral operational result enum.